go/vt/vtgate/engine/distinct.go

```diff
@@ -28,37 +28,37 @@ var _ Primitive = (*Distinct)(nil)
 
 // Distinct Primitive is used to uniqueify results
 type Distinct struct {
-	Source Primitive
+	Source        Primitive
+	ColCollations []collations.ID
```

Comment:
We can use a map over here which can simplify the access and also prevent any index out of range errors easily.

Reply:
when executing things in the runtime, we want them as fast as possible. A map is actually slower than accessing a position by offset in a slice, so I'll keep the collations as a slice here